### PR TITLE
updates for mee paper

### PR DESCRIPTION
**Pull request by sydnerecord**
- aligned wording with MEE submission text
- added circular plot example and bootstrap to univariate example
- updated text to reflect change from weight to mass for small mammal data

### CODE REVIEW

File: R/Ostats.R [hunk 1]
```diff
@@ -1,15 +1,16 @@
-#' @title Calculate O-statistics (community-level pairwise niche overlap statistics)
```
Comment (qdread) [inline R/Ostats.R:1]: Here I think the clarification in parentheses is still good to have because "O-statistic" is our jargon and it's good to define. We can still call them community-level statistics in the title even if we clarify later on that it need not be a "community" per se

File: R/Ostats_multivariate_plot.R [hunk 2]
```diff
@@ -1,11 +1,11 @@
-#' Plot multivariate community overlap
+#' Plot multivariate overlap
```
Comment (qdread) [inline R/Ostats_multivariate_plot.R:1]: Again I still do not see why we should delete the word community. It is confusing without. It is good to clarify later that it may be a population and not a community, but I still think it is good to generally say this is for community ecology analyses.

File: R/Ostats_plot.R [hunk 3]
```diff
@@ -1,10 +1,10 @@
-#'@title  Plot community trait overlap
+#'@title  Plot univariate trait overlap
```
Comment (qdread) [inline R/Ostats_plot.R:1]: See above

File: R/community_overlap.R [hunk 4]
```diff
@@ -1,8 +1,10 @@
-#' Community Overlap Calculation
+#' Community overlap Calculation
```
Comment (qdread) [inline R/community_overlap.R:1]: If overlap isn't capitalized, calculation should also not be

File: R/get_ses.R [hunk 5]
```diff
@@ -6,11 +6,11 @@
 #'
 #'@details This is an internal function not intended to be called directly.
 #'
-#'@seealso \code{\link{Ostats}} to calculate O-statistics (community-level
-#'  pairwise niche overlap statistics)
+#'@seealso \code{\link{Ostats}} to calculate O-statistics (pairwise niche overlap
```
Comment (qdread) [inline R/get_ses.R:9]: See above comment, should keep this as is

File: vignettes/Example1-Ostats-univariate.Rmd [hunk 6]
```diff
@@ -179,10 +179,10 @@ Regardless of whether the normalized or non-normalized overlap values are consid
...
 
 The input dataset needs to have these information: 
-* `plots` community or site identity: a vector of names to indicate which community or site the individual belongs to. 
+* `plots` community or site identity: a vector of names to indicate which community or site (i.e., population in a population-level analysis) the individual belongs to. 
 * `sp` taxon identification: a vector of species or taxa names.
 * `traits` trait measurements: a vector of trait measurements for each individual, or a matrix with rows representing individuals and columns representing traits.
-* `overlap_dat`  This input information is optional. It is an object containing the output of `Ostats` for the same data. If provided, it is used to label the plot panels with the community overlap values.
+* `overlap_dat`  This input information is optional. It is an object containing the output of `Ostats` for the same data. If provided, it is used to label the plot panels with the overlap values scaled up to be across species within a community or populations within a species.
```
Comment (qdread) [inline vignettes/Example1-Ostats-univariate.Rmd:185]: Not really "scaled up," more appropriate to say "averaged"

File: vignettes/Example1-Ostats-univariate.Rmd [hunk 7]
```diff
@@ -220,13 +222,121 @@ Ostats_plot(plots = siteID,
             traits = trait, 
             overlap_dat = small_mammal_Ostats, 
             use_plots = sites2use, 
-            name_x = expression(paste('log'[10], ' body mass (g)')), 
+            name_x = 'log10 Body Mass (g)',
```
Comment (qdread) [inline vignettes/Example1-Ostats-univariate.Rmd:225]: Please revert this change, I did that so that the plot we show in the journal would have a subscript for log10

File: vignettes/Example1-Ostats-univariate.Rmd [hunk 8]
```diff
@@ -220,13 +222,121 @@ Ostats_plot(plots = siteID,
...
+            circular = TRUE,
+            discrete = TRUE)
+```
+
+
+## Boostrapping
+
+### Example for generating bootstrap of linear univariate O-statistics
+
+This is an illustration of how to bootstrap O-statistics using the example data on small mammal body mass from Harvard Forest (HARV) and Jornada (JORN). It is written without any tidyverse packages, other than ggplot2, to not add any new dependencies.
```
Comment (qdread) [inline vignettes/Example1-Ostats-univariate.Rmd:259]: You can get rid of the sentence about not using tidyverse, I just said that for our own internal communication but I do not think it needs to be in the vignette

File: vignettes/Example2-Ostats-multivariate.Rmd [hunk 9]
```diff
@@ -22,19 +22,21 @@ library(Ostats)
 data(pitcher_traits)
 ```
 
+# Example 2: Multivariate Example - Pitcher plant leaf traits across populations
+
 # Overview
 
-This vignette on multi-variable community-level trait overlap statistics is intended for users who are already familiar with the single-variable O-statistics we introduce in the [first package vignette](Ostats-introduction.html).
+This vignette on multivariate trait overlap statistics is intended for users who are already familiar with the univariate O-statistics we introduce in the [first package vignette](Ostats-univariate.html).
```
Comment (qdread) [inline vignettes/Example2-Ostats-multivariate.Rmd:29]: See above comment, I think it is weird to delete all reference to community. Yes of course you "could" do this on data that is not an ecological community, but why are we getting rid of all mention of the word? It is still, to me, by far the most relevant use for this stuff

File: vignettes/Example1-Ostats-univariate.Rmd [hunk 10]
```diff
@@ -220,13 +222,121 @@ Ostats_plot(plots = siteID,
...
+
+# Convert observed normalized overlap to a data frame for plotting
+Ostats_observed <- with(Ostats_example, data.frame(siteID = dimnames(overlaps_norm)[[1]], overlap_norm = round(overlaps_norm[,1], 4)))
+```
+
+Define a custom function to do the bootstrap. Following the suggestion of Maitner et al. (2023), we sample with replacement from each species' traits within each community proportional to that species' relative abundance in that community. We use the nonparametric bootstrap approach. In other words, we are sampling directly from the raw data, instead of fitting a distribution and sampling from that, which would be parametric because we would be using the parameters of a fitted distribution to generate bootstrap samples.
+
+```{r}
+boot_fn <- function(i) {
+  # Bootstrap sample of same size as the data, within each community each row is sampled with replacement with equal probability
```
Comment (qdread) [inline vignettes/Example1-Ostats-univariate.Rmd:290]: These examples are great. Thanks for adding them. Here and elsewhere in this section, because you now have text in the vignette explaining what each piece of code does, we no longer need these long comment blocks in the code itself because it is redundant. So if there is any text in these comments that isn't in the text explanations between the code chunks, you could add it to the explanation then delete the code comments.

File: vignettes/Example1-Ostats-univariate.Rmd [hunk 11]
```diff
@@ -220,13 +222,121 @@ Ostats_plot(plots = siteID,
...
+  # This means the proportion of individuals of each species in the bootstrap sample will be similar to the observed data.
+  # Commented out version uses dplyr, the uncommented version uses base R code to get the same result.
```
Comment (qdread) [inline vignettes/Example1-Ostats-univariate.Rmd:292]: Lines 292-296 here should be deleted. We do not need them anymore, it was only my first attempt of writing this using dplyr code before I later went back and rewrote it using no external packages.

**Review by qdread — CHANGES_REQUESTED** [review]
Thanks for doing this! Sorry for all the nit-picky things but there are some minor things that I think need to be fixed before merging.

File: R/Ostats.R (hunk 1 above)
Comment (sydnerecord) [reply R/Ostats.R:1]: done

File: R/Ostats_multivariate_plot.R (hunk 2 above)
Comment (sydnerecord) [reply R/Ostats_multivariate_plot.R:1]: I added community back in.

File: R/Ostats_plot.R (hunk 3 above)
Comment (sydnerecord) [reply R/Ostats_plot.R:1]: added in community again

File: R/community_overlap.R (hunk 4 above)
Comment (sydnerecord) [reply R/community_overlap.R:1]: fixed

File: R/get_ses.R (hunk 5 above)
Comment (sydnerecord) [reply R/get_ses.R:9]: added community-level back in

File: vignettes/Example1-Ostats-univariate.Rmd (hunk 7 above)
Comment (sydnerecord) [reply vignettes/Example1-Ostats-univariate.Rmd:225]: done

File: vignettes/Example1-Ostats-univariate.Rmd (hunk 8 above)
Comment (sydnerecord) [reply vignettes/Example1-Ostats-univariate.Rmd:259]: deleted this last sentence referring to tidyverse

File: vignettes/Example1-Ostats-univariate.Rmd (hunk 10 above)
Comment (sydnerecord) [reply vignettes/Example1-Ostats-univariate.Rmd:290]: If it's ok, I like to keep the code annotated, too. I know it's redundant, but I think it's helpful for folks who are less familiar with R having taught a lot of students in the past who benefit from this sort of annotation.

File: vignettes/Example1-Ostats-univariate.Rmd (hunk 11 above)
Comment (sydnerecord) [reply vignettes/Example1-Ostats-univariate.Rmd:292]: Thanks for flagging this - I didn't catch it before.

File: vignettes/Example2-Ostats-multivariate.Rmd (hunk 9 above)
Comment (sydnerecord) [reply vignettes/Example2-Ostats-multivariate.Rmd:29]: added community back in
